Network/Mail/SMTP.hs

```diff
@@ -129,18 +132,24 @@ connectStream getMailHostName st = do
         msg <- tryCommand 3 (SMTPC st []) (HELO $ B8.pack senderHost) 250
         return (SMTPC st (tail $ B8.lines msg))
 
-parseResponse :: Handle -> IO (ReplyCode, ByteString)
-parseResponse st = do
+parseResponse :: Conn.Connection -> IO (ReplyCode, ByteString)
+parseResponse conn = do
     (code, bdy) <- readLines
     return (read $ B8.unpack code, B8.unlines bdy)
   where
+    getLines c acc = do
+      res <- Conn.connectionGetChunk c
+      if B8.null res
+         then return acc
+         else getLines c $ B8.append acc res
```

Comment:
This current implementation does not return a line, but instead returns the entire contents of the connection. Replacing it with `connectionGetLine` will fix it.